Vagrantfile

```diff
@@ -680,91 +644,27 @@ Vagrant.configure('2') do |config|
   # Provisioning
   #
   # Process one or more provisioning scripts depending on the existence of custom files.
-  #
-  # provison-pre.sh acts as a pre-hook to our default provisioning script. Anything that
-  # should run before the shell commands laid out in provision.sh (or your provision-custom.sh
-  # file) should go in this script. If it does not exist, no extra provisioning will run.
-  if File.exist?(File.join(vagrant_dir, 'provision', 'provision-pre.sh'))
-    config.vm.provision 'pre', type: 'shell', keep_color: true, path: File.join('provision', 'provision-pre.sh')
-  end
 
-  # provision.sh or provision-custom.sh
-  #
-  # By default, Vagrantfile is set to use the provision.sh bash script located in the
-  # provision directory. If it is detected that a provision-custom.sh script has been
-  # created, that is run as a replacement. This is an opportunity to replace the entirety
-  # of the provisioning provided by default.
-  if File.exist?(File.join(vagrant_dir, 'provision', 'provision-custom.sh'))
-    config.vm.provision 'custom', type: 'shell', keep_color: true, path: File.join('provision', 'provision-custom.sh')
-  else
-    config.vm.provision 'default', type: 'shell', keep_color: true, path: File.join('provision', 'provision.sh')
-  end
-
-  # Provision the dashboard that appears when you visit vvv.test
-  config.vm.provision 'dashboard',
-                      type: 'shell',
-                      keep_color: true,
-                      path: File.join('provision', 'provision-dashboard.sh'),
-                      args: [
-                        vvv_config['dashboard']['repo'],
-                        vvv_config['dashboard']['branch']
-                      ]
-
-  vvv_config['utility-sources'].each do |name, args|
-    config.vm.provision "utility-source-#{name}",
-                        type: 'shell',
-                        keep_color: true,
-                        path: File.join('provision', 'provision-utility-source.sh'),
-                        args: [
-                          name,
-                          args['repo'].to_s,
-                          args['branch']
-                        ]
+  ARGV.each do |arg|
+    match = arg.match(/(--provision-with=)(.*)/i);
+    if match
+      _prefix, req_prov = match.captures
+      config.vm.provision "#{req_prov}", type: 'shell', keep_color: true, path: File.join('provision', 'run-provisioners.sh'), args: [ req_prov ]
+    end
   end
 
+  config.vm.provision 'default', type: 'shell', keep_color: true, path: File.join('provision', 'run-provisioners.sh'), args: ( ENV.key?('VVV_PROVISION') ? ENV['VVV_PROVISION'] : [] )
```

Comment:
We shouldn't use a single file for all provisioners when using `vagrant`. Each provisioner needs to be its own provisioner

Reply:
What's the difference between having the VM be responsible for running the provisioners than to have `vagrant` be responsible for it?

Reply:
Because now it's not possible to run just a single provisioner without using a weird trick to manually parse the command line arguments. It's a huge code smell. We should just use normal provisioners and do things how they're supposed to be done. If we have 5 provisioners we should declare 5 provisioners.

As for the VM running provisioners, that's a different thing entirely, *Vagrant* is running the provisioners so we shouldn't mess with that. Having a single file to run the provisioners was meant to be used for testing and docker containers.